### ISSUE TITLE
Tracking Issue for Relicensing Effort

### ISSUE THREAD

**Issue by OneDeuxTriSeiGo** [pine64/OpenPineBuds#76]
This issue tracks all the separate tasks/issues involved in the relicensing effort. This issue was created separate from #18 as strictly speaking it's a subissue of issue #18.

Below is the status of all tasks involved in the relicensing effort:

- [x] [Source Analysis](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-analysis)
  - [x] [Upstream Analysis](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-analysis/blob/master/upstreams.rst)
  - [x] [Alternate Source Analysis](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-analysis/blob/master/alt-sources.rst)
  - [x] [Moved/Split Sources Identification](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-analysis/blob/master/moved-sources.rst)
  - [x] [Original Sources Identification](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-analysis/blob/master/original-sources.rst)
  - [x] [Contributor/Contributions Identification](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-analysis/tree/master/contributor-records)
- [ ] [Source Reconstruction](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-reconstruction): In Progress
  - [x] [Upstream Reconstruction](https://github.com/OneDeuxTriSeiGo/openpinebuds-source-reconstruction/tree/reconstruction/upstream)
  - [ ] Alternate Sources Reconstruction: In Progress
- [ ] Individual Contributor Relicensing: In Progress
  - [ ] #75: Opened. Pending Review.
  - [ ] @ shymega: pending approval to create issue.
  - [ ] @ ThatcherC: pending approval to create issue.
  - [ ] @ FintasticMan: pending approval to create issue.
  - [ ] @ frap129: pending approval to create issue.
  - [ ] @ Spagett1: pending approval to create issue.
  - [ ] @ pure5665: pending approval to create issue.
  - [ ] @ nik012003: pending approval to create issue
- [ ] BES Relicensing: Blocked on Source Reconstruction
  - [ ] Prepare minimal patchset of BES changes.
  - [ ] Prepare formal relicensing request.
  - [ ] Submit relicensing request to BES.
  - [ ] Ask BES about BT/BLE IP available in [OneDeuxTriSeiGo/device_soc_bestechnic:OpenHarmony-3.0-LTS](https://github.com/OneDeuxTriSeiGo/device_soc_bestechnic/tree/OpenHarmony-3.0-LTS)
- [ ] Relicensing Pull Request
  - [ ] Formalize `OPB-Source-Available` license.
  - [ ] add `LICENSES/` directory with all licenses included.
  - [ ] add accurate license declarations in each file.
  - [ ] add copyright attributions to each file, by contributor with DCO.
  - [ ] add reuse infrastructure.
- [ ] (moonshot) Identify path towards achieving full OpenChain ISO/IEC 5230 compliance.

**Comment by WebWire-NL** [comment]
Adding an input to this list, from the prebuilt side: the 23 `*.a` archives this build links were compiled with `-g`, so they still name their own sources. I pulled that metadata out (no decompilation, no vendor code, metadata only) and it gives a concrete target list:

- **498 source files** that this tree does not ship and that the archives name - the speech/ANC DSP (`services/multimedia/speech/src`, 130 files), the audio codecs (`services/multimedia/audio/codec`, 174, including FDK-AAC), the BLE host stack (`ble_stack/hl/src`, 34), LDAC/LHDC, and the BT protocol internals (btm/obex/hshf/besaud/map/pbap).
- **The exported surface**: 4,723 functions, of which 301 are declared by no header in the tree at all (180 external, 121 static) and 2,186 are private to the blobs.
- **The third-party boundary** that relicensing BES code cannot cover: Fraunhofer FDK-AAC (`fdkaac_codec`, 1,282 C++ symbols plus the `aacDecoder_*` API), Savitech LHDC and Sony LDAC.

Summary as a docs PR: #126. Full material, the 498-file list, a machine-readable inventory and the scripts that regenerate all of it: https://github.com/WebWire-NL/openpinebuds-integration (release v0.1.0).

If it helps, I can format the file list the way #92/#93 are laid out (per file, with declared licence, requested licence, and the diff against the reconstruction) so it can be folded into the existing requests.

— @WebWire-NL
